static/js/ReaderPanel.jsx

```diff
@@ -236,52 +236,26 @@ class ReaderPanel extends Component {
       highlightedRefsInSheet
     });
   }
-  showBaseText(ref, replaceHistory, currVersions={en: null, he: null}, filter=[]) {
+  showBaseText(ref, replaceHistory, currVersions={en: null, he: null}, filter=[], attemptConvertCommToBase=true) {
     // Set the current primary text `ref`, which may be either a string or an array of strings.
     // `replaceHistory` - bool whether to replace browser history rather than push for this change
     if (!ref) { return; }
     this.replaceHistory = Boolean(replaceHistory);
+    attemptConvertCommToBase = this.state.compare ? false : attemptConvertCommToBase;
     // console.log("showBaseText", ref, replaceHistory);
     if (this.state.mode === "Connections" && this.props.masterPanelLanguage === "bilingual") {
       // Connections panels are forced to be mono-lingual. When opening a text from a connections panel,
       // allow it to return to bilingual.
       this.state.settings.language = "bilingual";
     }
-    let refs, currentlyVisibleRef, highlightedRefs;
-    if (ref.constructor === Array) {
-      // When called with an array, set highlight for the whole spanning range
-      refs = ref;
-      currentlyVisibleRef = Sefaria.humanRef(ref);
-      let splitArray = refs.map(ref => Sefaria.splitRangingRef(ref));
-      highlightedRefs = [].concat.apply([], splitArray);
-    } else {
+    if (ref.constructor !== Array) {
       const oRef = Sefaria.parseRef(ref);
       if (oRef.book === "Sheet") {
         this.openSheet(ref);
         return;
       }
-      refs = [ref];
-      currentlyVisibleRef = ref;
-      highlightedRefs = [];
-    }
-
-    if (this.replaceHistory) {
-      this.props.saveLastPlace({ mode: "Text", refs, currVersions, settings: this.state.settings }, this.props.panelPosition);
     }
-    this.conditionalSetState({
-      mode: "Text",
-      refs,
-      filter,
-      currentlyVisibleRef,
-      currVersions,
-      highlightedRefs,
-      recentFilters: [],
-      menuOpen: null,
-      compare: false,
-      sheetID: null,
-      connectionsMode: "Resources",
-      settings: this.state.settings
-    });
+    this.props.openPanelAt(this.props.panelPosition, ref, currVersions, {settings: this.state.settings}, true, attemptConvertCommToBase, this.replaceHistory);
```

Comment:
I re-did this a bit.  You were right that I had not successfully converted all the functionality.  In particular, saveLastPlace was being funny: it was getting called twice when it should be called once.  I had tried to take saveLastPlace out of showBaseText and decided that was overcomplicating things.  To solve this, I added back into showBaseText the ability to saveLastPlace and added a parameter to openPanelAt 'saveLastPlace' that defaults to true.  The only case it is false is when showBaseText calls it as showBaseText has already called saveLastPlace.  Because of this change, the only thing I'm really removing from showBaseText is its ability to update its panel with its new ref, versions, and settings; openPanelAt now effectively does that by creating a new panel state based on the same properties.